### PR TITLE
Add some more lints for transmute

### LINKED ISSUE #2099: suggest char::from_u32 over transmute::<u32, char>
Thanks for the great talk @tomaka

### PR DESCRIPTION
**Pull request by pizzaiter**
Fixes #2099.

### CODE REVIEW

File: clippy_lints/src/transmute.rs [hunk 1]
```diff
@@ -76,11 +77,73 @@ declare_lint! {
...
+///
+/// **Why is this bad?** This might result in an invalid in-memory representation of a `bool`.
+///
+/// **Known problems:** None.
+///
+/// **Example:**
+/// ```rust
+/// let _: bool = std::mem::transmute(x) // where x: u8
+/// // should be:
+/// let _: bool = if x != 0 { true } else { false }
```
Comment (oli-obk) [inline clippy_lints/src/transmute.rs]: `let _: bool = x != 0;`

File: clippy_lints/src/transmute.rs [hunk 2]
```diff
@@ -177,6 +240,46 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for Transmute {
...
+                                    };
+                                    db.span_suggestion(e.span, "consider using", format!("std::char::from_u32({})", arg.to_string()));
+                                }
+                            ),
+                            (&ty::TyInt(ast::IntTy::I8), &ty::TyBool) |
+                            (&ty::TyUint(ast::UintTy::U8), &ty::TyBool) => span_help_and_lint(
+                                cx,
+                                TRANSMUTE_INT_TO_BOOL,
+                                e.span,
+                                &format!("transmute from a `{}` to a `bool`", from_ty),
```
Comment (oli-obk) [inline clippy_lints/src/transmute.rs]: Can you make this a `!= 0` suggestion?

**Comment by pizzaiter** [comment]
I updated the suggestion for the bool transmutes.

Should I rebase the branch to resolve the conflicts in the tests?

**Comment by pizzaiter** [comment]
Rebased

**Comment by pizzaiter** [comment]
:tada:

**Pull request merged**